### PR TITLE
feat: remove max af 1 from all workflows

### PR DESCRIPTION
**Pull request by mathiasbio**
### This PR:

This change has been made to remove the max allele frequency 1 filter, which has a risk of removing true somatic variants at a very high tumor purity. See issue: https://github.com/Clinical-Genomics/BALSAMIC/issues/1180

Removed: for now removed features.
- Removed bcftools command for setting soft-filter with maximum allele-frequency, which eventually leads to a hard removal of these variants. 

## Tests:

- [x] All analysis-types completed successfully
- [x] Few extra variants as a result of removing this filter

| sequencing type | T/TN         | case         | # PASS variants release 13 | # PASS variants this PR | # additional PASS variants | % extra PASS variants |
|-----------------|--------------|--------------|----------------------------|-------------------------|----------------------------|-----------------------|
| WGS             | T            | civilsole    |                      33051 |                   33051 |                          0 |                     0 |
| WGS             | T            | firstviper   |                      54072 |                   54072 |                          0 |                     0 |
| WGS             | TN           | fleetjay     |                       7189 |                    7190 |                          1 |                 0,01% |
| TGA             | T            | setamoeba    |                       2415 |                    2416 |                          1 |            0,04%           |
| TGA             | TN           | unitedbeagle |                       1573 |                    1573 |                          0 |                     0 |
| TGA UMI         | T (TNscope)  | equalbug     |                        158 |                     158 |                          0 |                     0 |
| TGA UMI         | T (VarDict)  | equalbug     |                         70 |                      70 |                          0 |                     0 |
| TGA UMI         | TN (TNscope) | uphippo      |                        124 |                     124 |                          0 |                     0 |
| TGA UMI         | TN (VarDict) | uphippo      |                        105 |                     105 |                          0 |                     0 |

Note that likely many variants would have been added in the T-only WGS cases if at the same time the T-only WGS specific filter `bcftools filter --threads {threads} --include 'FORMAT/ALT_F1R2 > {params.strand_reads[0]} && (FORMAT/ALT_F1R2 > 0  && FORMAT/ALT_F2R1 > {params.strand_reads[0]} &&  FORMAT/REF_F1R2 > {params.strand_reads[0]} && FORMAT/REF_F2R1 > {params.strand_reads[0]})' --soft-filter '{params.strand_reads[1]}' --mode '+' 
` was also removed. But as this filter also requires that some reads support a reference variant, then removing the MAX_AF 1 filter in this analysis type has no real effect. 

I think we can conclude with these stats that it is safe to remove the MAX AF filter. 

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: BALSAMIC/models/params.py [hunk 1]
```diff
@@ -214,7 +214,6 @@ class VarCallerFilter(BaseModel):
```
Comment (ivadym) [inline BALSAMIC/models/params.py:214]: Remove it also from the docstring :)

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_quality_filter.rule [hunk 2]
```diff
@@ -94,7 +90,6 @@ if config["analysis"]["sequencing_type"] == 'targeted' and config["analysis"]["a
       AD=[VARDICT.AD.tag_value, VARDICT.AD.filter_name],
       DP=[VARDICT.DP.tag_value, VARDICT.DP.filter_name],
       AF_min=[VARDICT.AF_min.tag_value, VARDICT.AF_min.filter_name],
-      AF_max=[VARDICT.AF_max.tag_value, VARDICT.AF_max.filter_name],
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/variant_calling/sentieon_quality_filter.rule:97]: We should probably remove it from `VARDICT_SETTINGS` too

**Review by ivadym — APPROVED** [review]
Looks good 💯 🎅

File: BALSAMIC/models/params.py (hunk 1 above)
Comment (mathiasbio) [reply BALSAMIC/models/params.py:213]: Nice catch!

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_quality_filter.rule (hunk 2 above)
Comment (mathiasbio) [reply BALSAMIC/snakemake_rules/variant_calling/sentieon_quality_filter.rule:97]: Ah yes, of course : )

**Pull request merged**